ordered_map.go

```diff
@@ -0,0 +1,42 @@
+package goavro
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+type KeyVal struct {
+	Key string
+	Val interface{}
+}
+
+// Define an ordered map
+type OrderedMap []KeyVal
+
+// Implement the json.Marshaler interface
+func (omap OrderedMap) MarshalJSON() ([]byte, error) {
```

Comment:
I think this is pretty good actually. I can't off the top of my head think of another way to preserve the ordering and still get what you want. I think this is pretty straightforward.